scout/adapter/mongo/query.py

```diff
@@ -126,29 +125,14 @@ def build_variant_query(
         if query.get("cohorts"):
             mongo_case_query["cohorts"] = {"$in": query["cohorts"]}
 
-        if mongo_case_query != {}:
-            mongo_case_query["collaborators"] = institute_id
+        if mongo_case_query != {} or institute_ids:
+            mongo_case_query["collaborators"] = {"$in": institute_ids}
             LOG.debug("Search cases for selection set, using query {0}".format(mongo_case_query))
             select_case_objs = self.case_collection.find(mongo_case_query)
             select_cases = [case_id.get("_id") for case_id in select_case_objs]
 
         if query.get("similar_case"):
-            similar_case_display_name = query["similar_case"][0]
```

Comment:
To simplify the complex code of this function, I've moved this code to another function (`_get_similar_cases` , see down)